### ISSUE TITLE
Ansible bottle not working

### ISSUE THREAD

**Issue by lucafavatella** [Homebrew/homebrew-core#12869]
- [x] Confirmed this is a problem with only one, specific formula and not Homebrew/brew? If it's a general Homebrew/brew problem please file this issue at https://github.com/Homebrew/brew/issues/new
- [x] Ran `brew update` and retried your prior step?
- [x] Ran `brew doctor`, fixed as many issues as possible and retried your prior step?
- [ ] Ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
```
$ brew gist-logs ansible
Error: No logs.
```
- [x] If `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?
```
$ brew config
HOMEBREW_VERSION: 1.1.13
ORIGIN: https://github.com/Homebrew/brew.git
HEAD: 46f6dc4c93b96abd528c3a0a808a5ca5a795c7f3
Last commit: 9 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: c1844902d996d22621481cda22812dfee78afe6b
Core tap last commit: 11 minutes ago
HOMEBREW_PREFIX: /Users/luca/homebrew
HOMEBREW_REPOSITORY: /Users/luca/homebrew
HOMEBREW_CELLAR: /Users/luca/homebrew/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://homebrew.bintray.com
CPU: quad-core 64-bit haswell
Homebrew Ruby: 2.0.0-p648
Clang: 8.0 build 800
Git: 2.12.2 => /Users/luca/homebrew/bin/git
Perl: /usr/bin/perl
Python: /Users/luca/homebrew/bin/python => /Users/luca/homebrew/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/bin/python2.7
Ruby: /usr/bin/ruby => /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby
Java: 1.8.0_121, 1.8.0_102, 1.8.0_92, 1.8.0_77, 1.8.0_74, 1.8.0_66
macOS: 10.11.6-x86_64
Xcode: N/A
CLT: 8.2.0.0.1.1480973914
X11: N/A

$ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry and just ignore them. Thanks!

Warning: Your Homebrew's prefix is not /usr/local.
You can install Homebrew anywhere you want but some bottles (binary packages)
can only be used with a /usr/local prefix and some formulae (packages)
may not build correctly with a non-/usr/local prefix.
```

To help us debug your issue please explain:
- What you were trying to do (and why)
  - Install ansible in order to use it.
- What happened (include command output)
  - ansible bottle poured but ansible not working.
```
$ ansible-playbook
Traceback (most recent call last):
  File "/Users/luca/homebrew/bin/ansible-playbook", line 21, in <module>
    from __future__ import (absolute_import, division, print_function)
ImportError: No module named __future__
$ head -n1 /Users/luca/homebrew/bin/ansible-playbook
#!/Users/luca/homebrew/Cellar/ansible/2.3.0.0_2/libexec/bin/python2.7
$ /Users/luca/homebrew/Cellar/ansible/2.3.0.0_2/libexec/bin/python2.7 -c "from __future__ import absolute_import"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: No module named __future__
$ find /Users/luca/homebrew -name __future__.py
/Users/luca/homebrew/Cellar/python/2.7.13/Frameworks/Python.framework/Versions/2.7/lib/python2.7/__future__.py
/Users/luca/homebrew/Cellar/python3/3.6.1/Frameworks/Python.framework/Versions/3.6/lib/python3.6/__future__.py
```
- What you expected to happen
  - ansible-playbook to run.
- Step-by-step reproduction instructions (by running `brew install` commands)

```
$ brew install ansible
$ ansible-playbook
```

----

See also https://github.com/Homebrew/brew/pull/2442

**Comment by dunn** [comment]
My guess is that this is because you installed Homebrew to a custom location.  @tdsmith does that sometimes cause problems for Python stuff?

**Comment by tdsmith** [comment]
Shouldn't; ;¯\\\_(ツ)\_/¯

**Comment by ilovezfs** [comment]
Yeah this is caused by the non-standard prefix. Apparently the `cellar :any` is inappropriate.

Using the bottle:
```
iMac-TMP:~ joe$ ansible-playbook
Traceback (most recent call last):
  File "/Users/joe/homebrew/bin/ansible-playbook", line 21, in <module>
    from __future__ import (absolute_import, division, print_function)
ImportError: No module named __future__
```

If I build it from source, it's fine.

**Comment by ilovezfs** [comment]
Even running `pip` in the virtualenv throws the error:
```
(libexec) iMac-TMP:2.3.0.0_2 joe$ pip
Traceback (most recent call last):
  File "/Users/joe/homebrew/Cellar/ansible/2.3.0.0_2/libexec/bin/pip", line 7, in <module>
    from pip import main
  File "/Users/joe/homebrew/Cellar/ansible/2.3.0.0_2/libexec/lib/python2.7/site-packages/pip/__init__.py", line 2, in <module>
    from __future__ import absolute_import
ImportError: No module named __future__
(libexec) iMac-TMP:2.3.0.0_2 joe$ 
```

**Comment by tdsmith** [comment]
This happens because we don't relocate .txt files and virtualenv stores the path to the python stdlib in `orig-prefix.txt`. Working on a PR.

**Comment by ilovezfs** [comment]
@tdsmith thanks Tim

**Comment by tdsmith** [comment]
This should have been fixed by #2598.
